fluid-constructs/src/main/java/me/escoffier/fluid/constructs/Data.java

```diff
@@ -0,0 +1,123 @@
+package me.escoffier.fluid.constructs;
+
+import java.util.*;
+
+/**
+ * Represents an item transiting on a stream.
+ * Instances of {@link Data} are immutable.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class Data<T> {
+
+  /**
+   * The payload, must not be {@code null}
+   */
+  private final T data;
+
+  /**
+   * A set of headers.
+   */
+  private final Map<String, Object> headers;
+
+  /**
+   * Creates a new instance of {@link Data}. This constructor does not set any headers (empty map).
+   *
+   * @param content the payload / item / content, must not be {@code null}
+   */
+  public Data(T content) {
+    this.data = Objects.requireNonNull(content);
+    this.headers = Collections.emptyMap();
+  }
+
+  /**
+   * Creates a new instance of {@link Data}. This constructor does not set any headers (empty map).
+   *
+   * @param content the payload / item / content
+   * @param headers the header, must not be {@code null}
+   */
+  public Data(T content, Map<String, Object> headers) {
```

Comment:
I'm wondering, can we try to use single name for Data "body"? Right now we have **payload**, **item** and **content**. Can we just pick one of those and use it everywhere? My personal favorite is **payload**.

Reply:
`Payload` is good. I picked `item` because it's the RX word. Gonna refactor.

Reply:
BTW, can you check whether you have enough data for the sinks to retrieve the headers and customize the dispatching?

Reply:
IMHO we have all we need for dispatching now since sinks receive fat event which could include all the metadata we want.